src/accelerate/utils/other.py

```diff
@@ -114,6 +114,35 @@ def save(obj, f):
         torch.save(obj, f)
 
 
+@contextmanager
+def clear_environment():
+    """
+    A context manager that will cache origin `os.environ` and replace it with a empty dictionary in this context.
+
+    When this context exits, the cached `os.environ` will be back.
+
+    Example:
+
+    ```python
+    >>> import os
+    >>> from accelerate.utils import clear_environment
+
+    >>> os.environ["FOO"] = "bar"
+    >>> with clear_environment():
+    ...     print(os.environ)  # prints "{}"
+    ...     os.environ["FOO"] = "new_bar"
+    ...     print(os.environ.get("FOO", None))  # prints "new_bar"
+    >>> print(os.environ["FOO"])  # prints "bar"
+    ```
```

Comment:
```suggestion
    >>> os.environ["FOO"] = "bar"
    >>> with clear_environment():
    ...     print(os.environ)
    "bar"
    ...     os.environ["FOO"] = "new_bar"
    ...     print(os.environ.get("FOO", None))
    "new_bar"
    >>> print(os.environ["FOO"])
    "bar"
    ```
```

Reply:
Thanks for your suggestion, I tried what your suggest, but unfortunately it would fail the doc style check.
I tried another way in my latest commit, you can have a look: [https://github.com/huggingface/accelerate/blob/a9c7cbb35aa0d0a7c736b1c41d1ef3712e43d792/src/accelerate/utils/other.py#L126-L141](https://github.com/huggingface/accelerate/blob/a9c7cbb35aa0d0a7c736b1c41d1ef3712e43d792/src/accelerate/utils/other.py#L126-L141)
Is it appropriate?

Reply:
Yep I'd say that is appropriate enough :) Good work!